doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1711,6 +1711,11 @@ Tactic notations allow customizing the syntax of tactics.
         - a global reference of term
         - :tacn:`unfold`
 
+      * - ``smart_global``
```

Comment:
Was it already supported and not documented or is this added by this PR? If that's the latter, then you might want to consider renaming it `smart_qualid`: farther from the implementation but closer to the documentation.

Reply:
Introduced by this PR.  We currently call `qualid` a `reference` in tactic-notation-land, though...  I feel like `smart_global` is less opaque than `smart_qualid`.  Are we planning to replace `reference` with `qualid` someday?  In any case, I'm happy to bow to consensus on this.

Reply:
> Are we planning to replace `reference` with `qualid` someday?

That would be the idea (it is already like this everywhere in the doc), but calling @herbelin to weigh on this.

Reply:
Hmm, to do this, do I change
```ocaml
let () = create_generic_quotation "smart_global" Pcoq.Prim.smart_global Stdarg.wit_smart_global
```
to
```ocaml
let () = create_generic_quotation "smart_qualid" Pcoq.Prim.smart_global Stdarg.wit_smart_global
```
or do I need to do more than this?

Reply:
#11840 was going in the direction of `smart_global`. The point is also that the section on tactic notations is as well on syntax as on semantics while the name `qualid` is mostly a syntax thing.

Additionally, a string is not technically a qualid (even though we could eventually think at allowing qualified strings such as `File.Mod."+"`).

Reply:
OK, then I guess we'll backtrack on the renaming in the doc. What about `smart_ref` / `smart_reference`? It's not clear immediately what `global` stands for (I guess it means "global reference") and I don't think it is used much in any user-facing doc / features so far. BTW, it looks like `ref` is supported there, not just `reference`. Do they mean the same thing?

Reply:
I'm happy to rename `smart_global` to `smart_reference`, or to make `reference` be smart and introduce `qualid` for what is currently `reference`, and change most things over to be (smart) `reference`. Would it be better to converge on #11840 first?  What's the status of #11840?

(And, yes, `ref` and `reference` are the same, I believe, see also #9122